app/models/user.rb

```diff
@@ -13,9 +13,11 @@ class User < ApplicationRecord
   has_many :friendships
   has_many :inverse_friendships, class_name: 'Friendship', foreign_key: 'friend_id'
 
-  # status = 1 --> Friends
-  # status = 0 --> Not Friends
-  # status = -1 --> Pending Friends
+  has_many :pending_friendships, -> { where(status: -1) }, class_name: 'Friendship', foreign_key: 'user_id'
+  has_many :pending_friends, through: :pending_friendships, source: :friend
+
+  has_many :incoming_friendships, -> { where(status: -1) }, class_name: 'Friendship', foreign_key: 'friend_id'
+  has_many :friend_requests, through: :incoming_friendships, source: :user
 
   def friends
```

Comment:
Now that you implemented the two-row friendships you can improve your `friends` method. Using enumerables to query the DB is not a good practice, you should make use of the associations you have instead. Use that Rails Active Record magic!